File: Easy/erogazkind/meta_erogazkinddefaultview/meta_erogazkinddefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_erogazkinddefaultview
{
    public class Meta_erogazkinddefaultview : Meta_easydata 
	{
        public Meta_erogazkinddefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "erogazkinddefaultview") {
				Name = "Tipologie di erogazione della didattica";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"iderogazkind"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "title", "Tipologia", nPos++);
						DescribeAColumn(T, "erogazkind_description", "Descrizione", nPos++);
						DescribeAColumn(T, "erogazkind_sortcode", "Ordinamento", nPos++);
						DescribeAColumn(T, "erogazkind_active", "Attivo", nPos++);
						DescribeAColumn(T, "erogazkind_ans", "Ans", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "erogazkind_sortcode";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/erogazkind/meta_erogazkinddefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_erogazkinddefaultview() {
        MetaData.apply(this, ["erogazkinddefaultview"]);
        this.name = 'meta_erogazkinddefaultview';
    }

    meta_erogazkinddefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_erogazkinddefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'title', 'Tipologia', null, 2000, 50);
						this.describeAColumn(table, 'erogazkind_description', 'Descrizione', null, 3000, 255);
						this.describeAColumn(table, 'erogazkind_active', 'Attivo', null, 4000, null);
						this.describeAColumn(table, 'erogazkind_ans', 'Ans', null, 5000, 10);
						this.describeAColumn(table, 'erogazkind_sortcode', 'Ordinamento', null, 6000, null);
						this.describeAColumn(table, 'erogazkind_title_en', 'Descrizione inglese', null, 9000, 50);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["iderogazkind"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					case "default": {
						return "title asc , erogazkind_sortcode desc";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('erogazkinddefaultview', new meta_erogazkinddefaultview('erogazkinddefaultview'));

	}());
